### PR TITLE
Beautify the dashboard

### PR DESCRIPTION
**Pull request by drairi**
From this:

Skærmbillede 2021-03-02 kl  21 16 38

To this:

Skærmbillede 2021-03-02 kl  21 13 13

I think we can all agree that's a lot prettier!

And it auto updates! 😍

### CODE REVIEW

**Comment by drairi** [comment]
NB. There will come a time when i'll want to upload a photo of each room, that's why there's such a large space. But before we do that, we'll need a database 😐

**Comment by rvedotrc** [comment]
Well there's already _sort of_ a database, insofar as tasks are saved to "var/tasks.json" :-p Lo-tech!

File: src/hue_api.ts [hunk 1]
```diff
@@ -47,3 +48,29 @@ export const request = (method, path, body) => new Promise((resolve, reject) =>
   req.write(JSON.stringify(body));
   req.end();
 });
+
+export const getRoomsFromGroups = (groups: Groups) => {
```
Comment (rvedotrc) [inline src/hue_api.ts]: This feels like it's maybe a level of functionality one level higher up than hue_api.ts. HueAPI is a pretty thin wrapper around the bridge, barely translating anything; getRoomsFromGroups feels a little more complex. Maybe should be elsewhere perhaps a new file. "dashboard.ts" maybe?

File: src/hue_api.ts [hunk 2]
```diff
@@ -47,3 +48,29 @@ export const request = (method, path, body) => new Promise((resolve, reject) =>
...
+  const rooms = [];
```
Comment (rvedotrc) [inline src/hue_api.ts]: Would be nice to type this more strongly:

```suggestion
  const rooms: Room[] = [];
```

and elsewhere
```
type Room = {
    id: string;
    name: string;
    ...;
}
```

File: public/js/app.js [hunk 3]
```diff
@@ -1,13 +1,14 @@
 let checkboxes = document.getElementsByClassName('on-off');
-let rooms = document.getElementsByClassName('room');
-let scenes = document.getElementsByClassName('scene');
+let roomOverlays = document.getElementsByClassName('room-overlay');
```
Comment (rvedotrc) [inline public/js/app.js]: Unused variable, according to my IDE

File: public/js/app.js [hunk 4]
```diff
@@ -17,17 +18,35 @@ var setScene = function() {
...
+  sceneImages[i].addEventListener('click', setScene, false);
 }
 
-for (var i = 0; i < scenes.length; i++) {
-  scenes[i].addEventListener('click', setScene, false);
+var updateState = function() {
+  var getStateCall = new XMLHttpRequest();
+  getStateCall.open("GET", "/groups/state", true);
+  getStateCall.onreadystatechange = function() {
+    if (this.readyState == 4 && this.status == 200) {
```
Comment (rvedotrc) [inline public/js/app.js]: I wonder if there's a constant (that evaluates to 4) to make this more readable?

**Review by rvedotrc — APPROVED** [review]

File: src/hue_api.ts (hunk 1 above)
Comment (drairi) [reply src/hue_api.ts]: A great idea! ❤️

File: src/hue_api.ts (hunk 2 above)
Comment (drairi) [reply src/hue_api.ts]: DONE!

File: public/js/app.js (hunk 3 above)
Comment (drairi) [reply public/js/app.js]: Correct, thank you! 😄

File: public/js/app.js (hunk 4 above)
Comment (drairi) [reply public/js/app.js]: Oh yeah, there is! `XMLHttpRequest.DONE` 😃

**Pull request merged**